packages/public/inspection-capture-web/src/PhotoCaptureHUD/PhotoCaptureHUD.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useRef, useState } from 'react';
+import { Sight } from '@monkvision/types';
+import { PhotoCaptureHUDButtons } from './PhotoCaptureHUDButtons';
+import { PhotoCaptureHUDPreviewAddDamage } from './PhotoCaptureHUDPreviewAddDamage';
+import { PhotoCaptureHUDPreview } from './PhotoCaptureHUDPreviewSight';
+import { usePhotoCaptureHUD } from './hook';
+
+export interface PhotoCaptureHUDProps {
+  sights?: Sight[];
+}
+
+export function PhotoCaptureHUD({ sights }: PhotoCaptureHUDProps) {
+  const [currentSight, setCurrentSight] = useState(sights?.[0]);
+  const [currentSightSliderIndex, setCurrentSightSliderIndex] = useState(0);
+  const [isOnAddDamage, setIsOnAddDamage] = useState<boolean>(false);
+  const [sightsTaken, setSightsTaken] = useState<Sight[]>([]);
+  const ref = useRef<HTMLDivElement>(null);
+
+  const handleOnSightSelected = (sight: Sight): void => {
+    setCurrentSight(sight);
+  };
+
+  const handleOnAddDamage = (state: boolean): void => {
+    setIsOnAddDamage(state);
+  };
+
+  const onScrollToSelected = (index: number, smooth: boolean): void => {
+    if (ref.current && ref.current.children.length > index) {
+      ref.current.children[index].scrollIntoView({
+        behavior: smooth ? 'smooth' : ('instant' as ScrollBehavior),
+        inline: 'center',
+      });
+      setCurrentSightSliderIndex(index);
+    }
+  };
```

Comment:
This "scrollToSelected" logic should be moved into the Slider component. It is the job of the Slider component to decide how and when to scroll its content.

Reply:
It was initially located down in the slider component, but I moved it up to the AddDamage component to enable scrolling to the last selected component when returning to the preview. Or should I create a other function which will be used in AddDamage return button?

Reply:
For me, your Slider component should have a useEffect that is called every time the currentSight changes that automatically slides the content of the slider. If you have this, normally when you come back from the Add Damage view, the slider component will be placed again in the DOM tree and the useEffect will be called automatically